core/types.go

```diff
@@ -385,12 +385,14 @@ func (s UnsignedDataSet) Clone() (UnsignedDataSet, error) {
 
 // SignedData is a signed duty data.
 type SignedData interface {
-	// Signature returns the signed duty data's signature.
-	Signature() Signature
-	// SetSignature returns a copy of signed duty data with the signature replaced.
-	SetSignature(Signature) (SignedData, error)
-	// MessageRoot returns the unsigned data message root.
-	MessageRoot() ([32]byte, error)
+	// Signatures returns the signed duty data's signatures.
+	Signatures() []Signature
+	// SetSignatures returns a copy of signed duty data with the signatures replaced.
+	SetSignatures([]Signature) (SignedData, error)
+	// MessageRoots returns the unsigned data message roots.
+	MessageRoots() ([][32]byte, error)
+	// HashRoot returns the final root of the unsigned data.
+	HashRoot() ([32]byte, error)
```

Comment:
Please rename this method as specified by the outcome of the Slack thread, as well as explaining what this field does a little more.

Reply:
Removed the `HashRoot()` interface method in favour of the function `HashMessageRoots(data)`, see updated code